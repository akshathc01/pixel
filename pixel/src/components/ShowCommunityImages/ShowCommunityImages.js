import React from 'react';

const ShowCommunityImages = ({ image }) => {
    return (
        <div data-testid="test-ShowCommunityImages" className="col-md-3 text-center m-3 p-3 bg-light text-info rounded shadow">
            <img className="rounded" style={{ height: '100%', width: '100%' }} src={image.url} alt={image.title} />
            <h3 className="mt-3 mb-2">{image.title}</h3>
        </div>
    );
};

export default ShowCommunityImages;