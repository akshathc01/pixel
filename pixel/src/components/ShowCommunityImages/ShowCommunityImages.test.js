import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ShowCommunityImages from './ShowCommunityImages'

afterEach(() => {
    cleanup()
})

test('test', () => {
    expect(true).toBe(true)
})

test('it will render ShowCommunityImages component', () => {
    const image = { _id: 1, imageName: '8c5dc61b7c71444ca37bc342020e7891.jpg', url: 'https://firebasestorage.googleapis.com/v0/b/pixel-image-uploader.appspot.com/o/images%2F8c5dc61b7c71444ca37bc342020e7891.jpg?alt=media&token=4fbf17bd-e125-4536-a02b-1f40518dbdcd', email: 'pixel@gmail.com', title: 'Sleeping Cat' }

    render(<ShowCommunityImages image={image} />)
    const showCommunityImagesElement = screen.getByTestId('test-ShowCommunityImages')
    expect(showCommunityImagesElement).toBeInTheDocument()
})

test('matches snapshot for ShowCommunityImages', () => {
    const image = { _id: 1, imageName: '8c5dc61b7c71444ca37bc342020e7891.jpg', url: 'https://firebasestorage.googleapis.com/v0/b/pixel-image-uploader.appspot.com/o/images%2F8c5dc61b7c71444ca37bc342020e7891.jpg?alt=media&token=4fbf17bd-e125-4536-a02b-1f40518dbdcd', email: 'pixel@gmail.com', title: 'Sleeping Cat' }

    const tree = renderer.create(<ShowCommunityImages image={image} />).toJSON()
    expect(tree).toMatchSnapshot()
})