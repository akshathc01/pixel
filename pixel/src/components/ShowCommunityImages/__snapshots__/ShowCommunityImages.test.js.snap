// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for ShowCommunityImages 1`] = `
<div
  className="col-md-3 text-center m-3 p-3 bg-light text-info rounded shadow"
  data-testid="test-ShowCommunityImages"
>
  <img
    alt="Sleeping Cat"
    className="rounded"
    src="https://firebasestorage.googleapis.com/v0/b/pixel-image-uploader.appspot.com/o/images%2F8c5dc61b7c71444ca37bc342020e7891.jpg?alt=media&token=4fbf17bd-e125-4536-a02b-1f40518dbdcd"
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
  <h3
    className="mt-3 mb-2"
  >
    Sleeping Cat
  </h3>
</div>
`;
